### PR TITLE
Deliverable 2

### PR DESCRIPTION
**Pull request by tkhquang**
# Mini-Challenge 2: Intro to Testing

## Answer the following questions first
1. Using `create-react-app`, what do we need to set up for testing?
2. What components are worth to test in your development?
3. Can you apply TDD once you already created components?

## Answers
1. Since `create-react-app` already ships with Jest, we only need to add `react-test-renderer` for rendering snapshots.
2. I think most components need to be tested, or at least add a test to make sure it renders correctly. Also, components which have complex inner logic should be well-tested with every edge case. Interactive/dynamic components should be tested carefully as well.
3. I think TDD does not really apply to maintenance, it focuses more on the development or implementation phase. As I understand, we write tests first (red phase), implement enough code to make them pass (green phase), the refactor the code to make it resemble some sensible design (refactor phase). So it doesn't apply to the context if we build the components first. In this case, I think we can only apply TDD when creating new or changing functionality of existing components. Still, I think create tests for already created components with already implemented functionalities is not TDD, because you skip the red phase.

Screen Shot 2021-06-28 at 20 06 53

### CODE REVIEW

**Comment by qtd-wizeline** [comment]
Please complete mini-challenge 3: Hooks for deliverable 2 also, thank you

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -12,16 +14,46 @@ import {
...
+  const { pathname } = useLocation();
+  const { search } = useYoutubeData();
+  const [query, setQuery] = useState('Wizeline');
+
+  const doSearch = useCallback(() => {
+    search({ query });
+  }, [search, query]);
+
+  const onSearchInputChange = (e) => {
+    setQuery(e.target.value);
```
Comment (qtd-wizeline) [inline src/components/Header/Header.jsx]: This will call the API whenever we change the input, better call the API after we stop typing for, e.g 500ms, or after pressing `Enter` key

Comment (tkhquang) [reply src/components/Header/Header.jsx]: @qtd-wizeline 
Edit: Thanks, I realize my mistake, I will fix this ASAP
